src/main/java/com/wasilewskyy/medical_clinic/controller/PatientController.java

```diff
@@ -21,7 +22,7 @@ public List<Patient> getAllPatients() {
     }
 
     @GetMapping("/{email}")
-    public Patient getPatientByEmail(@PathVariable String email) {
+    public Optional<Patient> getPatientByEmail(@PathVariable String email) {
```

Comment:
Controller raczej nie powinien zwracac Optionali -> poniewaz to ejst juz bezposrednie body, ktore uzytkownik dostaje wiec aplikacja powinna podjac decyzje jakie ono ma byc.

Optional jest tak naprawde wskazaniem ze pewnej danej moze nie byc i gdzies musimy wskazac co ma sie zadziać gdy tej danej nie ma -> i to powinno byc najczesciej w serwisie -> serwis powinien dla akutalnie wykonywanej funkcjonalnosci wskazac co ma sie zadziać kiedy jest zwrocony Optional np. z repozytorium.